### ISSUE TITLE
Compiling typescript code with webcontents event listener causes compilation error

### ISSUE THREAD

**Issue by Treverix** [electron/electron#23905]
### Preflight Checklist

* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
 electron 9.0.1, typescript 3.9.3
* **Operating System:**
  Windows 10
* **Last Known Working Electron version:**
 electron 8.x, typescript 3.8.x

### Expected Behavior
Valid code that uses documented API compiles with typescript

### Actual Behavior
The following line causes a typescript error, highlighting `new-window` on the IDE

```
this.browserWindow.webContents.on('new-window', (event: Event, url: string): void => { //... })
```

Displayed error:
```
TS2769: No overload matches this call.
   The last overload gave the following error
.     Argument of type '"new-window"' is not assignable to parameter of type '"zoom-changed"'.

electron.d.ts(9373, 5): The last overload is declared here.
```

**Comment by MarshallOfSound** [comment]
@Treverix Please provide a clonable minimal repro, that snippet you provided appears to work for me / typescript doesn't complain for Electron 7, 8, 9 or the nightly

**Comment by vladimiry** [comment]
Works for me this way

https://github.com/vladimiry/ElectronMail/blob/c1b2195797ffcd4332f0a5aa8a78df6db57b20de/src/electron-main/web-contents.ts#L70-L71

**Comment by Treverix** [comment]
While trying to reproduce on a boilerplate repro - and failing, I found the issue. 

The `new-window` listner (now?) requires an `Electron.NewWindowEvent` type while I declared the listener with an extended `Electron.Event`. So it searched for a matching overload of the `on` method but couldn't find one.

Interesting tough that is started failing after I upgraded electron and typescript as described above.
